stackslib/src/net/download/nakamoto.rs

```diff
@@ -0,0 +1,4283 @@
+        let sort_rc = sortdb
+            .pox_constants
+            .block_height_to_reward_cycle(sortdb.first_block_height, last_sort_height)
+            .expect("FATAL: burnchain tip is before system start");
+
+        let mut new_wanted_tenures = Self::load_wanted_tenures_at_tip(
+            self.last_sort_tip.as_ref(),
+            sort_tip,
+            sortdb,
+            &self.wanted_tenures,
+        )?;
+
+        let can_advance_wanted_tenures =
+            if let Some(prev_wanted_tenures) = self.prev_wanted_tenures.as_ref() {
+                !Self::have_unprocessed_tenures(
+                    sortdb
+                        .pox_constants
+                        .block_height_to_reward_cycle(
+                            sortdb.first_block_height,
+                            self.nakamoto_start_height,
+                        )
+                        .expect("FATAL: first nakamoto block from before system start"),
+                    &self.tenure_downloads.completed_tenures,
+                    prev_wanted_tenures,
+                    &self.tenure_block_ids,
+                    &sortdb.pox_constants,
+                    sortdb.first_block_height,
+                    invs.inventories.values(),
+                )
+            } else {
+                test_debug!("No prev_wanted_tenures yet");
+                true
+            };
+
+        if can_advance_wanted_tenures && self.reward_cycle != sort_rc {
+            let mut prev_wanted_tenures = vec![];
+            let mut cur_wanted_tenures = vec![];
+            let prev_wts = self.prev_wanted_tenures.take().unwrap_or(vec![]);
+            let cur_wts = std::mem::replace(&mut self.wanted_tenures, vec![]);
+
+            for wt in new_wanted_tenures
+                .into_iter()
+                .chain(prev_wts.into_iter())
+                .chain(cur_wts.into_iter())
+            {
+                test_debug!("Consider wanted tenure: {:?}", &wt);
+                let wt_rc = sortdb
+                    .pox_constants
+                    .block_height_to_reward_cycle(sortdb.first_block_height, wt.burn_height)
+                    .expect("FATAL: height before system start");
+                if wt_rc + 1 == sort_rc {
+                    prev_wanted_tenures.push(wt);
+                } else if wt_rc == sort_rc {
+                    cur_wanted_tenures.push(wt);
+                } else {
+                    test_debug!("Drop wanted tenure: {:?}", &wt);
+                }
+            }
+
+            prev_wanted_tenures.sort_by(|wt1, wt2| wt1.burn_height.cmp(&wt2.burn_height));
+            cur_wanted_tenures.sort_by(|wt1, wt2| wt1.burn_height.cmp(&wt2.burn_height));
+
+            test_debug!("prev_wanted_tenures is now {:?}", &prev_wanted_tenures);
+            test_debug!("wanted_tenures is now {:?}", &cur_wanted_tenures);
+
+            self.prev_wanted_tenures = if prev_wanted_tenures.len() > 0 {
```

Comment:
Should replace with 
`!prev_wanted_netures.is_empty()`

Reply:
Sure